test/unit/Cover/editCover.js

```diff
@@ -298,6 +384,111 @@ describe('editCover', function () {
     });
   });
 
+  it('should allow to reduce period and increase amount', async function () {
+    const { cover } = this;
+
+    const [coverBuyer] = this.accounts.members;
+
+    const { productId, coverAsset, period, amount, targetPriceRatio, priceDenominator } = coverBuyFixture;
+
+    await buyCoverOnOnePool.call(this, coverBuyFixture);
+
+    const {
+      expectedPremium,
+      segment,
+      coverId: expectedCoverId,
+    } = await buyCoverOnOnePool.call(this, { ...coverBuyFixture, period: period * 2 });
+
+    const expectedRefund = segment.amount
+      .mul(targetPriceRatio)
+      .mul(segment.period)
+      .div(MAX_COVER_PERIOD)
+      .div(priceDenominator);
+
+    const expectedEditPremium = expectedPremium; // reduced period by half and increased amount to double
+    const extraPremium = expectedEditPremium.sub(expectedRefund);
```

Comment:
Given we intend to have half the period and double the amount - the refund should cover the new amount perfectly and I'd expect zero extra premium (unless you get on the surge, which you should avoid in this test).